applications/GeoMechanicsApplication/custom_constitutive/incremental_linear_elastic_interface_law.cpp

```diff
@@ -0,0 +1,131 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Wijtze Pieter Kikstra
+//                   Anne van de Graaf
+//
+
+#include "incremental_linear_elastic_interface_law.h"
+#include "custom_geometries/line_interface_geometry.h"
+#include "geo_mechanics_application_constants.h"
+#include "geo_mechanics_application_variables.h"
+
+namespace Kratos
+{
+
+ConstitutiveLaw::Pointer GeoIncrementalLinearElasticInterfaceLaw::Clone() const
+{
+    return std::make_shared<GeoIncrementalLinearElasticInterfaceLaw>(*this);
+}
+
+ConstitutiveLaw::SizeType GeoIncrementalLinearElasticInterfaceLaw::WorkingSpaceDimension()
+{
+    return 2;
```

Comment:
Could this ever become 3d (e.g. when using it for a interface plane, where you have multiple 'tangential' directions and a normal)?

Reply:
I can imagine that at some point we will extend this class towards 3D models. In that case, I would suggest to supply the desired dimension on construction of an instance of this class, and store it with the object. Then this query should return whatever value the corresponding data member contains. Do you think this might work, or do you foresee any issues there?

Reply:
I don't see any issues in this particular function, I think the challenge is more in the formulation, since then there is more than one 'SHEAR' direction. I don't know enough to see if this way of formulating it, makes it harder to go to 3D at some point.